### PR TITLE
93 editor buttons and 95 cartoons social buttons

### LINKED ISSUE #93: CMS: move open editor and save buttons into editor.pug file
This issue involves some refactoring. We want to move the buttons created in the `createButtons` function in `public/assets/ts/admin/editor.ts` and define them as buttons in `generic-editor.pug.` This also means we have to move out all `onclick` functionality and refactor that in the editor file.

Screenshot 2023-02-07 at 8 31 27 PM

**Acceptance Critera**
* The "Open Markdown Editor" and "Save [ file ]' Buttons are part of `generic-editor.pug`
* All current interactivity with the buttons is preserved.

### LINKED ISSUE #95: Cartoons - About - Buttons fix
On widths of 1300px - 1500px, the cartoons pink buttons don't display cleanly.

image

**Acceptance criteria:**
* These buttons don't experience visual bugs on 1300px - 1500px
* The buttons change collectively - we no longer have a situation where only 3/4 buttons appear differently from the last one.

### PR DESCRIPTION
**Pull request by ad-world**
Closes #93 
Closes #95

### CODE REVIEW

File: public/assets/ts/admin/editor.ts [hunk 1]
```diff
@@ -23,38 +23,21 @@ class Editor {
...
+    const saveButton = document.getElementById("save-editor-btn");
 
-    const openButtonContainer = document.createElement("div");
-    openButtonContainer.classList.add("open-editor-container");
-    openButtonContainer.appendChild(openButton);
-    container.appendChild(openButtonContainer);
-
-    const saveButton = document.createElement("button");
-    saveButton.classList.add("save", "pink-button");
+    openButton.innerHTML = "Open Markdown Editor";
```
Comment (some-molasses) [inline public/assets/ts/admin/editor.ts:37]: Would it be possible to move the contents of the button into the pug file?

For the `saveButton`, we could move `Save <span class="editor-name"></span>` into pug and change the value of the span from JS.

File: views/pages/admin/generic-editor.pug [hunk 2]
```diff
@@ -31,6 +31,10 @@ block body
     div(id="body")
       h2(id="editor-title")=editorName
       div(id="jsoneditor" data-source=editorSource)
+      div(class="open-editor-container")
+        button(class="pink-button disabled open-editor-button" id="open-editor-btn" disabled="true")
```
Comment (some-molasses) [inline views/pages/admin/generic-editor.pug:35]: Since we're using the `pink-button`s, can we use the `+pink-button()` mixin instead?

**Review by some-molasses — CHANGES_REQUESTED** [review]
thanks for taking these on!

File: views/pages/admin/generic-editor.pug (hunk 2 above)
Comment (ad-world) [reply views/pages/admin/generic-editor.pug:35]: Pink button doesn't support extra classes or attributes like "disabled." I could just set the extra stuff in the ts file, but then we have another case of basically created a button in ts

Comment (some-molasses) [reply views/pages/admin/generic-editor.pug]: sounds good!  I might rework the pink button later to be able to handle that functionality, but this works for now

**Review by some-molasses — APPROVED** [review]

File: views/pages/admin/generic-editor.pug (hunk 2 above)
Comment (ad-world) [reply views/pages/admin/generic-editor.pug]: thanks! i wanted to try n tackle it, but didn't want to break the usage of the pink button in multiple places

**Pull request merged**